### ISSUE TITLE
More Detailed Instructions

### ISSUE THREAD

**Issue by RushHour2k5** [MnrGreg/homebridge-ecovacs-deebot#1]
Can you please provide more detailed instructions on how to setup your plugin. I re-compiled the .go file into an executable on my Ubuntu system but when running it I get the following messages.

"/home/deebot/deebot status                                      
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/skburgart/go-vacbot.LoadConfiguration(0x6be0c5, 0x1a, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/deebot/go/src/github.com/skburgart/go-vacbot/vacbot.go:129 +0x1d9
github.com/skburgart/go-vacbot.NewFromConfigFile(0x6be0c5, 0x1a, 0xfb)
	/home/deebot/go/src/github.com/skburgart/go-vacbot/vacbot.go:48 +0x50
main.main()
	/home/deebot/go/src/github.com/MnrGreg/homebridge-ecovacs-deebot/deebot.go:67 +0x1b0"

Not sure if this is normal or if I'm running into an issue as it's late and I cannot start the Vacuum.

**Comment by RushHour2k5** [comment]
Ok, I've gotten a bit further with my Ecovac DEEBOT OZMO 930 however it still appears to fail now with the STARTTLS handshake. Any advice?

administrator@hb-svr1-sbc:~$ /home/deebot/deebot charge
2019/02/17 02:43:34 login successful
2019/02/17 02:43:35 get auth code successful
2019/02/17 02:43:36 get user access token successful
2019/02/17 02:43:37 starttls handshake: x509: certificate has expired or is not yet valid

**Comment by dpauken** [comment]
I haven't even gotten a valid login. Perhaps I'm misunderstanding the config steps. what should i be passing for deviceID? do i put the hashed values into the json file, or is that done on the fly?

**Comment by RushHour2k5** [comment]
@dpauken the only thing I entered for a hashed value is an MD5 of my password. DeviceID I used the serial number but now the issue is that it keeps referring to the TLS error which seems related to the XMPP portion of the plugin which I cannot for the life of me figure out.
Issue for "starttls handshake: x509: certificate has expired or is not yet valid" within @skburgart's GoLang script that is tied into this plugin was also reported by @MnrGreg can be found here. https://github.com/skburgart/go-vacbot/issues/3
It appears related to the XMPP portion of the script based on the issue found at https://github.com/wpietri/sucks/issues/62 in which they removed pyasn1 and pyasn1-modules from Python however cannot find how to do this within the GoLang script of the XMPP.go file.
